jsx/Form.js

```diff
@@ -630,6 +630,73 @@ var DateElement = React.createClass({
   }
 });
 
+/**
+ * Time Component
+ * React wrapper for a <input type="time"> element.
+ */
+var TimeElement = React.createClass({
+
+  propTypes: {
+    name: React.PropTypes.string.isRequired,
+    label: React.PropTypes.string,
+    value: React.PropTypes.string,
+    id: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    required: React.PropTypes.bool,
+    onUserInput: React.PropTypes.func
+  },
+
+  getDefaultProps: function() {
+    return {
+      name: '',
+      label: '',
+      value: '',
+      id: null,
+      disabled: false,
+      required: false,
+      onUserInput: function() {
+        console.warn('onUserInput() callback is not set');
+      }
+    };
+  },
+  handleChange: function(e) {
+    this.props.onUserInput(this.props.name, e.target.value);
+  },
+  render: function() {
+    var disabled = this.props.disabled ? 'disabled' : null;
+    var required = this.props.required ? 'required' : null;
+    var requiredHTML = null;
+
+    // Add required asterix
+    if (required) {
+      requiredHTML = <span className="text-danger">*</span>;
+    }
+
+    return (
+      <div className="row form-group">
+        <label className="col-sm-3 control-label" htmlFor={this.props.label}>
+          {this.props.label}
+          {requiredHTML}
+        </label>
+        <div className="col-sm-9">
+          <input
+            type="time"
+            className="form-control"
+            name={this.props.name}
+            id={this.props.label}
+            onChange={this.handleChange}
+            value={this.props.value || ""}
+            required={required}
+            disabled={disabled}
+            pattern="([0-1][0-9]|2[0-4]|[1-24]):([0-5][0-9])(:([0-5][0-9]))?"
```

Comment:
I don't think the `[1-24]` part is right.

Reply:
I put that there in case they wanted to simply put 1:00 instead of 01:00. But we can force the second option if you think that's best.

Reply:
I've also given them the option of inserting seconds after minutes, but that might not be necessary either.